### ISSUE TITLE
[6.0.0-rc.4] Unsuccessful attempt at using with immutable

### ISSUE THREAD

**Issue by vitaminwater** [erikras/redux-form#1513]
Hi,

Sorry for the un-helpful title, this is more to summarize my attempt at using v6 with immutable.

First things first here is (part of) the code (the rest mostly comes from [react-boilerplate](https://github.com/mxstbr/react-boilerplate)):

**reducers.js:**
```
import {reducer as formReducer} from 'redux-form/immutable';

[...]

export default function createReducer(asyncReducers) {
  return combineReducers({
    [...]
    form: formReducer,
    [...]
  });
}

```

**Login.js**

```

/*
 *
 * Login
 *
 */

import React from 'react';
import { compose } from 'redux';
import { connect } from 'react-redux';
import selectLogin from './selectors';
import styles from './styles.css';

import { reduxForm, Field } from 'redux-form/immutable';

import { Grid, FormGroup, FormControl, ControlLabel, HelpBlock, Button } from 'react-bootstrap';

const validate = values => {
  const errors = {};
  if (!values.email) {
    errors.email = 'Veuillez entrer votre email';
  } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.email)) {
    errors.email = 'Adresse email non valide';
  }

  if (!values.password) {
    errors.password = 'Veuillez entrer votre mot de passe';
  }
  console.log(errors);
  return errors;
};

class InputField extends React.Component {
  render() {
    const { input, label, placeholder, type, meta: { touched, error } } = this.props;

    return (<FormGroup validationState={touched && error ? 'error' : 'success'}>
      <ControlLabel>{placeholder}</ControlLabel>
      <FormControl {...input} placeholder={label} type={type} />
      {touched && error && <HelpBlock>{error}</HelpBlock>}
    </FormGroup>);
  }
};

class Login extends React.Component { // eslint-disable-line react/prefer-stateless-function

  _handleSubmit() {
    const { fields: {email, password} } = this.props;
    this.props.login(email.value, password.value);
  }

  render() {
    const { handleSubmit } = this.props;
    return (
      <Grid className={styles.login}>
        <form onSubmit={handleSubmit}>
          <Field name='email' type='email' component={InputField} placeholder='email' />
          <Field name='password' type='password' component={InputField} placeholder='mot de passe' />
          <Button type='submit'>
            Valider
          </Button>
        </form>
      </Grid>
    );
  }
}

const mapStateToProps = selectLogin();

function mapDispatchToProps(dispatch) {
  return {
    dispatch,
    login(email, password) {
      dispatch({type: 'LOGIN', payload: {email, password}});
    },
  };
}

Login = reduxForm({
  form: 'login',
  validate,
  //getFormState: (state) => state.get('form').toJS(),
})(Login);

export default compose(
  connect(mapStateToProps, mapDispatchToProps),
)(Login);

```

I first tried to install it through npm:
`npm install --save redux-form@v6.0.0-rc.4`

No errors on the terminal, but the browser (chrome) console gives:
> createPropTypes.js:5 Uncaught TypeError: Cannot read property 'PropTypes' of undefined

Which is strange because createPropTypes.js is not even present in the v6.0.0-rc.4 codebase...
And a quick `grep createPropTypes -R node_modules/redux-form/*` neither shows any occurence to the file nor the function.

Then I tried cloning it to use it as a linked npm module:
```
git clone git@github.com:erikras/redux-form.git
cd redux-form
git checkout v6
npm i
npm link

```

then in my project directory:
```
npm link redux-form
```

Which first gives in the terminal:

> Module build failed: Error: Couldn't find preset "react-hmre" relative to directory "/Users/stant/Documents/zilia/internals/redux-form"

`npm install react-hmre` in redux-form fixes it.

Now the browser console shows:

`Super expression must either be null or a function` in SubmissionError.js

It appears the the import is doing something strange not returning the default module export...
(Dirty-)fixed it with a 

` class SubmissionError extends ExtendableError.default {`
instead of
` class SubmissionError extends ExtendableError {`

Now my page renders, but, in the browser console, I get a:

> invariant.js:38 Uncaught Invariant Violation: addComponentAsRefTo(...): Only a ReactOwner can have refs. You might be adding a ref to a component that was not created inside a component's `render` method, or you have multiple copies of React loaded (details: https://fb.me/react-refs-must-have-owner).

Still digging through it. Send help;)

**Comment by hathawsh** [comment]
I'm not sure about that first error you got, but I can say that redux-form doesn't work well with `npm link`. Actually, I've never had a good experience with `npm link` with any package. @erikras has said in the past that instead of using `npm link`, he relies heavily on unit tests and he sometimes copies the `lib` folder of a built `redux-form` project into the `node_modules` of another project.

**Comment by javatask** [comment]
+1

**Comment by erikras** [comment]
Yes, `npm link` has never worked well for me with this library or any other.

**Comment by erikras** [comment]
Fix published in [`v6.0.3`](https://github.com/erikras/redux-form/releases/tag/v6.0.3).

**Comment by nhducit** [comment]
I still got this issue with redux-form 6.2 and react-boilerplate 3.3.1
